src/mailchimp/mailchimp_manager.py

```diff
@@ -60,8 +76,13 @@ def ping(self) -> Response:
         Raises:
             TODO: Detail potential errors
         """
-        raise NotImplementedError
-    
+        try:
+                self._client.set_config(self.keys)
+                response = client.ping.get()
+                return response
+        except ApiClietError:
+		raise ApiClientError
```

Comment:
would you like me to not raise any errors and if an error happens the program throws it at the user.
        self._client.set_config(self.keys)
        response = client.ping.get()
        return response
should this be the code. I cannot raise the error without using an if/else statement or a try/except block

Reply:
Basically yes. If a function does not catch an error via a `try`-`except` block, it will automatically be raised to the calling function. Elizabeth's code will catch the error.

Reply:
Except that `ping()` should not be setting the server and API/token. That line should only be in `set_authorisation()`.